test/export/dashboard.test.ts

```diff
@@ -0,0 +1,159 @@
+                    expect(options.method).toBe("POST");
+                    expect(JSON.parse(options.body as string)).toEqual({
+                        dashboardExport: {
+                            dashboardUri: "/gdc/md/testProjectId/obj/dashboard123",
+                            filters: [
+                                {
+                                    dateFilter: {
+                                        type: "relative",
+                                        from: "-11",
+                                        to: "11",
+                                        granularity: "GDC.time.month",
+                                    },
+                                },
+                                {
+                                    attributeFilter: {
+                                        displayForm: "/gdc/md/testProjectId/obj/700",
+                                        negativeSelection: false,
+                                        attributeElements: [
+                                            "/gdc/md/testProjectId/obj/750",
+                                            "/gdc/md/testProjectId/obj/751",
+                                        ],
+                                    },
+                                },
+                                {
+                                    dateFilter: {
+                                        type: "absolute",
+                                        from: "2019-08-06",
+                                        to: "2019-08-08",
+                                        granularity: "GDC.time.month",
+                                    },
+                                },
+                            ],
+                        },
+                    });
+                });
+        });
+
+        it("should return pdf uri", () => {
+            return dashboardModuleMock()
+                .exportDashboardToPdf(projectId, dashboardUri, [], { pollStep: 1 })
+                .then((result: IExportResponse) => {
+                    expect(result).toEqual({
+                        uri: "/gdc/exporter/result/testProjectId/pdfFileUri",
+                    });
+                });
+        });
+    });
+
+    describe("failed pdf export", () => {
+        it("should return error when send export request fail", () => {
+            fetchMock.post(endpoint, {
+                status: BAD_REQUEST_STATUS,
+            });
+
+            return dashboardModuleMock()
+                .exportDashboardToPdf(projectId, dashboardUri, [])
+                .then(null, (error: ApiResponseError) => {
+                    // error thrown in xhr.ts
+                    expect(error.response.status).toEqual(400);
+                    expect(error.message).toEqual("Bad Request");
+                });
+        });
+
+        it("should return error when polling fail", () => {
+            jest.setTimeout(5000);
```

Comment:
Why do we need timeout here?

Reply:
In `mockPollingRequestWithStatus`, it takes more than 3000ms to return `finishedTask`, so we call `jest. setTimeout` to overwrite default timeout of running single test, make it wait for longer before ending it.

Reply:
are there any way to speed it up?
waiting for 5s could make our tests slower

Reply:
Yes, you're right.
I have tried several numbers, but I get the default one in Jest eventually, it is 5s, 4s is still failed.
By default, Jest sets 5s for us, but test case gets failed until I set it `jest.setTimeout(5000);` explicitly.
I guess 5s default is for all `describe`, not single test.

Reply:
Althought the timeout is 5s, but the test could finish ASAP base on setTimeout/deplay in our code ;)

Reply:
I'm kind of still confused... why are you not using `jest.useFakeTimers()` or something else useful from https://jestjs.io/docs/en/timer-mocks? and I would say there could be another way how to mock fetch to return immediately bad request.

But using jest.setTimeout() looks to me like a bad idea which makes test potentially unstable...

Reply:
Maybe something like this ? https://github.com/gooddata/gooddata-js/blob/master/test/report.test.ts#L220